crates/ruff/resources/test/fixtures/flake8_bugbear/B031.py

```diff
@@ -78,6 +78,29 @@ def collect_shop_items(shopper, items):
     for shopper in shoppers:
         collect_shop_items(shopper, section_items)  # B031
 
+for _section, section_items in itertools.groupby(items, key=lambda p: p[1]):
+    # Mutually exclusive branches shouldn't trigger the warning
+    if _section == "greens":
+        collect_shop_items(shopper, section_items)
+    elif _section == "frozen items":
+        collect_shop_items(shopper, section_items)
+    else:
+        collect_shop_items(shopper, section_items)
+    # Now, it should detect
+    collect_shop_items(shopper, section_items)  # B031
```

Comment:
We may be missing "multiple usages within the same arm, top-level":

```py
for _section, section_items in itertools.groupby(items, key=lambda p: p[1]):
    match _section:
        case "greens":
            collect_shop_items(shopper, section_items)
            collect_shop_items(shopper, section_items)
```

This seems like it should trigger B031, but doesn't.

Reply:
Yes, this should be triggered, let me take a look at that.